src/plots/plots.js

```diff
@@ -2848,7 +2849,50 @@ function _transition(gd, transitionOpts, opts) {
     return transitionStarting.then(function() { return gd; });
 }
 
+var typedArrays = {
+    int8: typeof Int8Array !== 'undefined' ? Int8Array : null,
+    uint8: typeof Uint8Array !== 'undefined' ? Uint8Array : null,
+    uint8clamped: typeof Uint8ClampedArray !== 'undefined' ? Uint8ClampedArray : null,
+    int16: typeof Int16Array !== 'undefined' ? Int16Array : null,
+    uint16: typeof Uint16Array !== 'undefined' ? Uint16Array : null,
+    int32: typeof Int32Array !== 'undefined' ? Int32Array : null,
+    uint32: typeof Uint32Array !== 'undefined' ? Uint32Array : null,
+    float32: typeof Float32Array !== 'undefined' ? Float32Array : null,
+    float64: typeof Float64Array !== 'undefined' ? Float64Array : null,
+    bigint64: typeof BigInt64Array !== 'undefined' ? BigInt64Array : null,
+    biguint64: typeof BigUint64Array !== 'undefined' ? BigUint64Array : null
+};
+
+function _decode(cont) {
+    if(cont.dtype && cont.bvals) {
+        var T = typedArrays[cont.dtype];
+        if(T) {
+            return new T(b64.decode(cont.bvals));
+        }
+    }
+
+    for(var prop in cont) {
+        if(prop[0] !== '_' && cont.hasOwnProperty(prop)) {
+            var item = cont[prop];
+            if(Lib.isPlainObject(item)) {
+                var r = _decode(item);
+                if(r !== undefined) cont[prop] = r;
+            }
+        }
+    }
+}
+
+function decodeB64Arrays(gd) {
+    for(var i = 0; i < gd._fullData.length; i++) {
+        _decode(gd._fullData[i]);
```

Comment:
Can we be sure that it's not itself an (encoded) ndarray?

Reply:
Could you explain how an encoded ndarray looks like?
Is there a way (or an npm package) to unpack an encoded ndarray in JavaScript?

Reply:
Sorry for not being clear; I meant an array, encoded as you proposed in this PR with `{shape, dtype, bvals}` :)

Reply:
Have you encountered a bug with current setting?

Reply:
No, it's just that if an array is passed to `_decode()` it will return the "decoded" version, and this is somewhat easy to overlook, so I wanted to make sure that we know this is not the case here. Sorry for the noise ...

Reply:
It is actually a good point to consider. Thanks for your contribution @almarklein.

Reply:
I think this is fine. `gd._fullData[i]` and `gd._fullLayout` will never legitimately be arrays.  And by this point we already through with processing default values and so I think the `shape`/`dtype`/`bvals` would be stripped out already.